File: Python/complement-of-base-10-integer.py
# Time:  O(logn)
# Space: O(1)

class Solution(object):
    def bitwiseComplement(self, N):
        """
        :type N: int
        :rtype: int
        """
        mask = 1
        while N > mask:
            mask = mask*2+1
        return mask-N

File: C++/complement-of-base-10-integer.cpp
// Time:  O(logn)
// Space: O(1)

class Solution {
public:
    int bitwiseComplement(int N) {
        int X = 1;
        while (N > X) {
            X = X * 2 + 1;
        }
        return X - N;
    }
};
